src/transformers/models/bart/modeling_bart.py

```diff
@@ -1013,18 +1015,20 @@ def forward(
         if input_ids is not None and inputs_embeds is not None:
             raise ValueError("You cannot specify both decoder_input_ids and decoder_inputs_embeds at the same time")
         elif input_ids is not None:
-            input_shape = input_ids.size()
+            input = input_ids
+            input_shape = input.shape
```

Comment:
It looks like `input_shape` is not use in this `forward` anymore, and the other variables are not strictly necessary either, so you could should remove those to elif and just do an
`elif input_ids is None and inputs_embeds is  None` for the last `ValueError`.

Reply:
`input_shape` is used for the `_prepare_decoder_attention_mask` and `_expand_mask` methods within this `forward`.

Reply:
Ah yes, was looking at the wrong forward 🤦‍♂️